server/sonar-server/src/main/java/org/sonar/ce/http/CeHttpClient.java

```diff
@@ -45,8 +45,8 @@
 
   private final File ipcSharedDir;
 
-  public CeHttpClient(Settings props) {
-    this.ipcSharedDir = new File(props.getString(PROPERTY_SHARED_PATH));
+  public CeHttpClient(Configuration config) {
+    this.ipcSharedDir = new File(config.get(PROPERTY_SHARED_PATH).get());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)